.github/workflows/argocd_githash_deploy.yml

```diff
@@ -0,0 +1,159 @@
+name: Githash-Based ArgoCD Deployment Workflow
+description: >
+  This workflow will update the ArgoCD config's image tag with
+  the given docker image tag, and either push directly to the
+  specified branch OR create a pull request describing the changes
+  between the previous version and new version.
+
+  This workflow assumes that your argocd config resides on a branch
+  _without_ branch protection rules.
+
+  The PR feature will push to a branch containing the environment
+  name and the app name. If a PR already exists with the same name,
+  changes will be force pushed. This prevents PRs from piling up.
+
+  The PR also contains a changelog. Changes are detected using
+  `git log`. This is why your docker image tagging strategy needs
+  to be git hash based, otherwise this step will fail.
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+        description: >
+          The name of the target environment. This is used
+          for informational purposes only. No logic is determined
+          by the environment name.
+
+      app-name:
+        required: true
+        type: string
+        description: >
+          The name of the application being updated. This is used
+          for informational purposes only. No logic is determined
+          by the app name.
+
+      argocd-branch:
+        required: false
+        type: string
+        default: argocd
+        description: >
+          The name of the branch where the ArgoCD config resides.
+          The default value is 'argocd'. This branch cannot have
+          branch protection rules on it if the `make-pr` input is
+          false.
+
+      make-pr:
+        required: false
+        type: boolean
+        default: true
+        description: >
+          Describes whether to push directly to the branch or
+          make a PR. If this value is false, an SSH_KEY secret
+          must be provided.
+
+      docker-image-tag:
+        required: true
+        type: string
+        description: >
+          The git hash of the latest docker image tag. We will
+          be updating ArgoCD / EKS to this version.
+
+      argocd-config-location:
+        required: true
+        type: string
+        description: >
+          The relative path of the ArgoCD configuration file.
+
+      argocd-config-key:
+        required: false
+        type: string
+        default: image.tag
+        description: >
+          The name/yq-path of the key being updated. This defaults
+          to 'image.tag'.
+
+      runs-on:
+        required: false
+        type: string
+        default: ubuntu-latest
+        description: >
+          The name of the environment to run on. This defaults to
+          'ubuntu-latest'
+
+    secrets:
+      SSH_KEY:
```

Comment:
does this need to be GIT_WRITER_SSH_KEY to match line 106 below?

Reply:
ahh good catch, ill change line 106